src/build/supportedExtensions.js

```diff
@@ -144,7 +144,7 @@ exports.extensions = {
     { icon: 'phpunit', extensions: ['phpunit.xml'], filename: true },
     { icon: 'plantuml', extensions: ['pu', 'plantuml', 'iuml', 'puml'] },
     { icon: 'procfile', extensions: ['procfile'], filename: true },
-    { icon: 'prolog', extensions: ['P', 'pro'], languages: [languages.prolog] },
```

Comment:
LanguageIds can be added via plugins. In the case of `Prolog` you can find the extension [here](https://marketplace.visualstudio.com/items?itemName=rebornix.prolog) and if you see [their contribution point ](https://github.com/rebornix/vscode-prolog/blob/master/package.json) you'll see that the id of prolog was correct.